File: backend/app/schemas/tag.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class TagRead(BaseModel):
    id: int
    key: str
    name: str
    category: str
    description: str | None
    color: str | None
    status: str
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)


class CustomerTagRead(BaseModel):
    id: int
    customer_id: int
    tag: TagRead
    source: str
    status: str
    evidence: list[dict]
    created_by: int | None
    confirmed_by: int | None
    created_at: datetime
    confirmed_at: datetime | None

File: frontend/src/types/tag.ts
import type { AIEvidence } from './ai'

export interface Tag {
  id: number
  key: string
  name: string
  category: string
  description: string | null
  color: string | null
  status: string
  created_at: string
  updated_at: string
}

export interface CustomerTag {
  id: number
  customer_id: number
  tag: Tag
  source: string
  status: string
  evidence: AIEvidence[]
  created_by: number | null
  confirmed_by: number | null
  created_at: string
  confirmed_at: string | null
}
